activiti-api-impl/activiti-api-process-runtime-impl/src/main/java/org/activiti/runtime/api/model/impl/APIProcessInstanceConverter.java

```diff
@@ -47,27 +42,11 @@ public ProcessInstance from(org.activiti.engine.runtime.ProcessInstance internal
         //To do: it is not the best way to search parentProcessId by this method, it will require extra queries!
         
         //Set parent ProcessInstance Id
-        if(internalProcessInstance.getSuperExecutionId()!=null && ExecutionEntity.class.isInstance(internalProcessInstance)) {
+        if(internalProcessInstance.getSuperExecutionId()!=null) {
             ExecutionEntity executionEntity = ExecutionEntity.class.cast(internalProcessInstance);
-            
-            if (Context.getCommandContext()==null) {
-                  ProcessEngineConfigurationImpl impl = (ProcessEngineConfigurationImpl)
-                        ProcessEngines.getDefaultProcessEngine().getProcessEngineConfiguration();
-
-                        impl.getCommandExecutor().execute(new Command<Void> () {
-
-                        @Override
-                        public Void execute(CommandContext commandContext) {
-                            Optional.ofNullable(executionEntity.getSuperExecution())
-                                    .ifPresent(superExecution -> processInstance.setParentId(superExecution.getProcessInstanceId()));
-
-                             return null;
-                        }});
-
-            } else {
-                  Optional.ofNullable(executionEntity.getSuperExecution())
-                          .ifPresent(superExecution -> processInstance.setParentId(superExecution.getProcessInstanceId()));
-            }
```

Comment:
@erdemedeiros @CTI777 I have removed the hack in APIProcessInstanceConverter.java above in favor of using Mybatis mapping to eagerly fetch superExecution entity via association property. It seems like we always need to resolve parent process instance attributes in event converters and queries, so I think it makes sense to always fetch it.

See Mabits changes below: https://github.com/Activiti/Activiti/pull/2260/commits/f3543483fbb600e1083f42b138d2c000c226ef2e#diff-b9f50b6ee46df36fb61073916d78de78 for details.